### ISSUE TITLE
query_df returning unexpected value when the columns are using Date32 and date before 1970-01-01

### ISSUE THREAD

**Issue by pyemma91** [ClickHouse/clickhouse-connect#254]
### Describe the bug
Hi, my team is working on a migration from SF to CH and we encountered an issue when using `query_df` to read columns of Date32 data type. The API would return incorrect value if the date is before 1970-01-01. We have checked that the data stored within CH is right, and use `query` API to read the raw results is also correct.

### Steps to reproduce
1. Create a test table using Nullable(Date32)
2. Use the query_df to read the table

### Expected behaviour
The returned result should match the result stored in DB

### Code example
```SQL
DROP TABLE IF EXISTS TEST_DATE32;

CREATE TABLE TEST_DATE32 (
    `timestamp` Nullable(Date32),
    `event_id` UInt8
)
ENGINE = ReplicatedMergeTree
ORDER BY ();

INSERT INTO TEST_DATE32 VALUES ('2001-01-01', 1), ('1969-01-01', 2);

SELECT * FROM TEST_DATE32
```

```python
import clickhouse_connect

def getClickhouseConnection(db, env):
    return clickhouse_connect.create_client(
        username=env.get('CLICKHOUSE_DATABASE_USER') if env else os.getenv('CLICKHOUSE_DATABASE_USER'),
        password=env.get('CLICKHOUSE_DATABASE_PWD') if env else os.getenv('CLICKHOUSE_DATABASE_PWD'),
        host=env.get('CLICKHOUSE_DATABASE_HOST') if env else os.getenv('CLICKHOUSE_DATABASE_HOST'),
        port = 443,
        interface='https',
        secure=True,
        database=db,
        settings = {
            'mutations_sync': 2,
            'replication_alter_partitions_sync': 2
        }
    )

def readTabelCH(table, schema, env=None):
    with getClickhouseConnection(schema, env) as chConn:
        query = f"SELECT * FROM {table}"
        df = chConn.query_df(query)
        return df

readTabelCH("TEST_DATE32", "<YOUR_DB_NAME>", env=env)
```

The result returned would be
<img width="279" alt="Screenshot 2023-10-20 at 9 18 08 PM" src="https://github.com/ClickHouse/clickhouse-connect/assets/137432439/a54595d6-d12b-4f20-84d8-3eecd785689f">

#### Environment
* clickhouse-connect version: 0.6.6
* Python version: 3.8.16
* Operating system: MacOS 13.6 (22G120)

#### ClickHouse server
* ClickHouse Server version:
* ClickHouse Server non-default settings, if any:
* `CREATE TABLE` statements for tables involved:
```
CREATE TABLE TEST_DATE32 (
    `timestamp` Nullable(Date32),
    `event_id` UInt8
)
ENGINE = ReplicatedMergeTree
ORDER BY ();
```
* Sample data for these tables, use [clickhouse-obfuscator](https://github.com/ClickHouse/ClickHouse/blob/master/programs/obfuscator/Obfuscator.cpp#L42-L80) if necessary
```
INSERT INTO TEST_DATE32 VALUES ('2001-01-01', 1), ('1969-01-01', 2);
```

**Comment by genzgd** [comment]
Thanks for the report!  This should be fixed in the latest 0.6.17 release, it was an incorrect use of an unsigned numpy type.

**Comment by pyemma91** [comment]
> Thanks for the report! This should be fixed in the latest 0.6.17 release, it was an incorrect use of an unsigned numpy type.

Thanks for the reply, will upgrade to new version and try again.
